src/Stratis.Bitcoin/P2P/PeerConnector.cs

```diff
@@ -166,33 +168,48 @@ public void AddPeer(NetworkPeer peer)
             this.ConnectedPeers.Add(peer);
         }
 
+        /// <summary>Determines whether or not a connector can be started.</summary>
+        internal abstract bool OnCanStartConnectAsync { get; }
+
+        /// <summary>Specific peer connector initialization for each concrete implementation of this class.</summary>
+        internal abstract void OnInitialize();
```

Comment:
You could potentially achieve the same result by making `Initialize` itself virtual, just a suggestion.

Reply:
I want to force the concrete implementations to implement OnInitialize() so I made it abstract :)

Reply:
Ah yes valid point